sql/core/src/main/scala/org/apache/spark/sql/execution/aggregate/udaf.scala

```diff
@@ -469,7 +471,8 @@ case class ScalaAggregator[IN, BUF, OUT](
   with ImplicitCastInputTypes
   with Logging {
 
-  private[this] lazy val inputDeserializer = inputEncoderNR.resolveAndBind().createDeserializer()
+  // input encoder is resolved by ResolveEncodersInScalaAgg
+  private[this] lazy val inputDeserializer = inputEncoder.createDeserializer()
   private[this] lazy val bufferEncoder =
     agg.bufferEncoder.asInstanceOf[ExpressionEncoder[BUF]].resolveAndBind()
```

Comment:
shall we do the same thing for `bufferEncoder`? we may need to put buffer encoder as a parameter of `ScalaAggregator`

Reply:
The main reason inputEncoder is a parameter is because `Aggregator` doesn't have inputEncoder as a member, but I'll verify buffer and output in the unit testing

Reply:
output encoder is OK, as it's only used to covert external scala value to internal catalyst value, no need to resolve it.